src/run_tribler_headless.py

```diff
@@ -129,7 +147,11 @@ def main(argv):
     parser.add_argument('--help', '-h', action='help', default=argparse.SUPPRESS,
                         help='Show this help message and exit')
     parser.add_argument('--statedir', '-s', default=None, help='Use an alternate statedir')
-    parser.add_argument('--restapi', '-p', default=-1, type=int, help='Use an alternate port for REST API')
+    parser.add_argument('--downloaddir', default=None, help='Use an alternative download directory')
```

Comment:
i have changed downloaddir to snake case. Anything else you would like changed?
I'm a bit uncertain if the headless code vs --core should both exist or be merged somehow.

Reply:
Thank you!

I noticed another detail that I overlooked in the previous review.

We actually have an existing method to pass the REST API port to the core, specifically through the `CORE_API_PORT` environment variable.

For instance:
```shell
export CORE_API_PORT=20100
python src/run_tribler.py
```

I acknowledge that it might seem inconsistent to have some variables passed via environment settings, some through the config, and others as arguments. However, this is currently the way it's set up in Tribler.

For the API Port, I believe it would be best to maintain consistency by using the approach found in `run_tribler.py`.

https://github.com/Tribler/tribler/blob/f141dccfba52d253d0bb0863e4d2374b0045c713/src/run_tribler.py#L88

Alternatively, a hybrid approach might also be suitable:

```python
http_port = options.restapi_http_port or os.environ.get('CORE_API_PORT', -1)
if http_port > 0:
    config.api.http_port = http_port
```

Beyond this observation, I have no concerns regarding the Python part of this PR. 👍

> I'm a bit uncertain if the headless code vs --core should both exist or be merged somehow.

I'm not certain, but this seems beyond the scope of the current PR.

Reply:
I added direct support for CORE_API_PORT & removed the parameter from the docker-compose file.